include/deal.II/base/parallel.h

```diff
@@ -107,7 +111,7 @@ namespace parallel
 
   /**
    * An algorithm that performs the action <code>*out++ =
-   * predicate(*in++)</code> where the <code>in</code> iterator ranges over
+   * function(*in++)</code> where the <code>in</code> iterator ranges over
```

Comment:
There are a number of places in this file that used the word "predicate" instead of "function object". This was uniformly wrong in this file: a "predicate" is a function that returns a `bool`, but the ones in this file do generally not. I fixed this while here, both in the documentation, in variable names, and in template type names.